airgun/entities/discoveredhosts.py

```diff
@@ -0,0 +1,223 @@
+        :returns the entity table row columns values
+        """
+        view = self.navigate_to(self, 'All')
+        wait_for(
+            lambda: view.is_searchable(),
+            fail_condition=False,
+            timeout=300,
+            delay=10,
+            fail_func=view.browser.refresh,
+            logger=view.logger
+        )
+        result = wait_for(
+            lambda: view.search('name = "{0}"'.format(entity_name)),
+            fail_condition=[],
+            timeout=300,
+            delay=10,
+            logger=view.logger
+        )
+        return result.out[0]
+
+    def search(self, value):
+        """Search for 'value' and return discovery hosts values that match.
+
+        :param str value: filter text.
+        """
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def read(self, entity_name):
+        """Return a dict with properties of discovered host."""
+        view = self.navigate_to(self, 'Details', entity_name=entity_name)
+        return view.read()
+
+    def provision(self, entity_name, host_group, organization, location,
+                  quick=True, host_values=None):
+        """Provision a discovered host with name entity_name."""
+        if host_values is None:
+            host_values = {}
+
+        view = self.navigate_to(self, 'Details', entity_name=entity_name)
+        view.actions.fill('Provision')
+        dialog_view = DiscoveredHostProvisionDialog(self.browser)
+        dialog_view.fill(dict(
+            host_group=host_group,
+            organization=organization,
+            location=location,
+        ))
+        if quick:
+            dialog_view.quick_create.click()
+        else:
+            dialog_view.customize_create.click()
+            discovered_host_edit_view = DiscoveredHostEditProvisioningView(
+                self.browser)
+            discovered_host_edit_view.fill(host_values)
+            discovered_host_edit_view.submit.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def auto_provision(self, entities_list):
+        """Auto provision discovered hosts which names supplied in
+        entities_list.
+        """
+        if not isinstance(entities_list, (list, tuple)):
+            entities_list = [entities_list]
+        view = self.navigate_to(
+            self, 'Auto Provision', entities_list=entities_list)
+        view.submit.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def reboot(self, entities_list):
+        """Reboot discovered hosts which names supplied in entities_list."""
+        if not isinstance(entities_list, (list, tuple)):
+            entities_list = [entities_list]
+        view = self.navigate_to(
+            self, 'Reboot', entities_list=entities_list)
+        view.submit.click()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def assign_organization(self, entities_list, organization):
```

Comment:
Just curious - why different from host's implementation of helpers/navigate steps/views, as the situation is kinda similar to me?
https://github.com/SatelliteQE/airgun/pull/179/files#diff-a4f1f5c4c861bb6e90ce95c2c5c915c7R71

Reply:
we implemented in hosts that way as host has too many actions, here in this PR we have a restraint number, if it we can make an arrangement that all entities action should be that way, no problem I'll update this PR, but we should take into consideration that all entities that has actions (except host) are implemented as this PR.